examples/robot_example.cpp

```diff
@@ -0,0 +1,94 @@
+
+// #include "kodlab_mjbots_sdk/robot_control_loop.h"
+// #include "kodlab_mjbots_sdk/behavior.h"
+// #include "kodlab_mjbots_sdk/robot_super.h"
+#include <vector>
+#include <iostream>
+#include <memory>
+#include "kodlab_mjbots_sdk/common_header.h"
+#include "ManyMotorLog.hpp"
+#include "ModeInput.hpp"
+#include "kodlab_mjbots_sdk/robot_interface.h"
+#include "kodlab_mjbots_sdk/mjbots_control_loop.h"
+
+#include "examples/simple_robot.h"
+
+class SimpleRobotControlLoop : public kodlab::mjbots::MjbotsControlLoop<ManyMotorLog, ModeInput, SimpleRobot>
+{
+    using MjbotsControlLoop::MjbotsControlLoop;
+
+    void CalcTorques() override
+    {
+        robot_->Update();
+    }
+    void PrepareLog() override
+    {
+        for (int servo = 0; servo < num_motors_; servo++)
+        {
+            log_data_.positions[servo] = robot_->GetJointPositions()[servo];
+            log_data_.velocities[servo] = robot_->GetJointVelocities()[servo];
+            log_data_.modes[servo] = static_cast<int>(mjbots_interface_->GetJointModes()[servo]);
+            log_data_.torques[servo] = robot_->GetJointTorqueCmd()[servo];
+        }
+        for (int servo = num_motors_; servo < 13; servo++)
+        {
+            log_data_.positions[servo] = 0;
+            log_data_.velocities[servo] = 0;
+            log_data_.modes[servo] = 0;
+            log_data_.torques[servo] = 0;
+        }
+    }
+
+    void ProcessInput() override
+    {
+        robot_->mode = lcm_sub_.data_.mode;
+        std::cout << "Switching to behavior " << robot_->mode << std::endl;
+        // If the kill robot mode is detected kill robot using CTRL_C flag handler.
+        if (robot_->mode == robot_->KILL_ROBOT)
+        { // KILL_ROBOT
+            kodlab::CTRL_C_DETECTED = true;
+        }
+    }
+};
+
+template <class joint_type>
+class JointSharedVector
+{
+    static_assert(std::is_base_of<JointBase, joint_type>::value);
+    std::vector<std::shared_ptr<joint_type>> v_;
+
+public:
+    template <typename... Args>
+    void addJoint(Args... args)
+    {
+        v_.push_back(std::make_shared<joint_type>(args...));
+    }
+    operator std::vector<std::shared_ptr<joint_type>>() { return v_; }
+};
+
+int main(int argc, char **argv)
+{
+
+    // Setup joints
+    // std::vector<kodlab::mjbots::JointMoteus> joints;
+    // joints.emplace_back(100, 4, 1, 0,   1, 0);
+    // joints.emplace_back(101, 4,-1, 0, 5.0/3.0, 0);
+
+    JointSharedVector<kodlab::mjbots::JointMoteus> joints;
```

Comment:
Some of the simplifications that happened in the later commits mean this new class while useful as mentioned above may no longer be necessary.

Reply:
If it still provides some useful functionality, just stick it in its own header file to be included as needed.

Reply:
Done. Also leaving both implementations in the example.